### PR TITLE
Clean unit test cli output

### PR DESCRIPTION
**Pull request by theodur**
## Description
This branch is intended to clean the unit test output.

## Testing done
Tested locally.

## Screenshots
**Before:**
```
․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․,․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․target Element not found
․․․․․Warning: Failed prop type: The prop `removeInProgressForm` is marked as required in `FormStartControls`, but its value is `undefined`.
    in FormStartControls
Warning: Failed prop type: The prop `prefillAvailable` is marked as required in `FormStartControls`, but its value is `undefined`.
    in FormStartControls
․․․Warning: Failed prop type: The prop `fetchInProgressForm` is marked as required in `FormStartControls`, but its value is `undefined`.
    in FormStartControls
Warning: Failed prop type: The prop `startPage` is marked as required in `FormStartControls`, but its value is `undefined`.
    in FormStartControls
․Warning: Failed prop type: The prop `formSaved` is marked as required in `FormStartControls`, but its value is `undefined`.
    in FormStartControls
․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
```

**After:**

Screen Shot 2020-02-16 at 11 45 36 PM

## Acceptance criteria
- [ ] Warnings are not shown while running unit tests
- [ ] All unit tests pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by rianfowler** [comment]
@theodur: 
What does this look like when a unit test fails? 

@bkjohnson 
Is there a way we can see what this looks like on Circle? I'm not sure how something like the nyan reporter will work on their UI.

**Comment by theodur** [comment]
@rianfowler Here is what it looks like when there is an error:

Screen Shot 2020-02-17 at 9 58 43 AM

If the nyan reporter doesn't work in circle ci, I can change it to something more simplistic like the [progress](https://mochajs.org/#progress), [landing strip](https://mochajs.org/#landing-strip) or [min](https://mochajs.org/#min) reporter.

**Comment by bkjohnson** [comment]
It might work in Circle, but right now Circle is overriding that and using a junit reporter to get XML coverage reports.  https://app.circleci.com/jobs/github/department-of-veterans-affairs/vets-website/688/parallel-runs/0/steps/0-106

**Comment by cvalarida** [comment]
image
This is what it looks like in CircleCI.

File: .circleci/config.yml [hunk 1]
```diff
@@ -75,6 +91,7 @@ workflows:
   test_and_build:
     jobs:
       - lint
+      - nyan-report
       - unit-test
```
Comment (cvalarida) [inline .circleci/config.yml]: Looks like this will run the unit tests twice, right? Do we need that?

**Review by cvalarida — COMMENTED** [review]
Looks good! Only concern I have is with regards to running the unit tests twice.

cc @U-DON since you worked on the `run-isolated-tests.sh` script that the `unit-test` job runs. Is there any reason we would need that separate `nyan-report` job?

File: .circleci/config.yml (hunk 1 above)
Comment (theodur) [reply .circleci/config.yml]: That commit was made for the purpose of seeing how the output would look on circle ci. We don't need it, I'll revert it.

File: src/platform/testing/unit/mocha-setup.js [hunk 2]
```diff
@@ -26,6 +26,12 @@ export default function setupJSDom() {
   //   throw new Error('Refusing to override existing document and window.');
   // }
 
+  // Prevent warnings from displaying
+  /* eslint-disable no-console, func-names */
+  console.error = function() {};
+  console.warn = function() {};
```
Comment (U-DON) [inline src/platform/testing/unit/mocha-setup.js]: Does declaring these as arrow functions avoid the linter from complaining about `func-names`?

Comment (theodur) [reply src/platform/testing/unit/mocha-setup.js]: Can't remember why I didn't have them as arrow functions in the first place. I just converted them.

**Review by cvalarida — APPROVED** [review]
I think this is fine. If we needed to use the `console.error()` and `console.warn()` for real in a unit test, I think we'll have to fall back to `process.stderr.write()`, which isn't ideal, but also not the end of the world.

In the Node REPL, I tried re-importing `console` to see if that would solve it, but no dice, it would seem. Not sure if that would change in an actual Node _module_, but I won't hold my breath.
image

**Comment by theodur** [comment]
`console.log` and `console.info` should still be able to work in unit tests, which also isn't ideal if someone wants to use`console.warn` or `console.error`, but it's the next best thing. Wish there was another way to suppress the warning errors.

**Pull request merged**